harness/determined/exec/tensorboard.py

```diff
@@ -1,143 +1,250 @@
 import json
+import logging
 import os
 import subprocess
 import sys
 import time
+import uuid
 from typing import Callable, List, Tuple
+from urllib.parse import urlparse
 
 import boto3
 import requests
+from requests.exceptions import ConnectionError, HTTPError
 
+TENSORBOARD_TRIGGER_READY_MSG = "TensorBoard contains metrics"
+EXP_CONFIG_JSON_PATH = "/run/determined/workdir/experiment_config.json"
 
-def set_s3_region(bucket: str) -> None:
-    endpoint_url = os.environ.get("DET_S3_ENDPOINT", None)
+logger = logging.getLogger("determined.exec.tensorboard")
+formatter = logging.Formatter("%(levelname)s - %(message)s")
+stderr_handler = logging.StreamHandler(sys.stderr)
+stderr_handler.setFormatter(formatter)
+logger.addHandler(stderr_handler)
 
-    client = boto3.client("s3", endpoint_url=endpoint_url)
-    bucketLocation = client.get_bucket_location(Bucket=bucket)
+logger.setLevel(logging.DEBUG)
 
-    region = bucketLocation["LocationConstraint"]
 
-    if region is not None:
-        # We have observed that in US-EAST-1 the region comes back as None
-        # and if AWS_REGION is set to None, tensorboard fails to pull events.
-        print(f"Setting AWS_REGION environment variable to {region}.")
-        os.environ["AWS_REGION"] = str(region)
+def get_tensorboard_args(tb_version, tfevents_dir, add_args):
+    """Build tensorboard startup args.
 
+    Args are added and deprecated at the mercy of tensorboard; all of the below are necessary to
+    support versions 1.14, 2.4, and 2.5
+
+    - Tensorboard 2+ no longer exposes all ports. Must pass in "--bind_all" to expose localhost
+    - Tensorboard 2.5.0 introduces an experimental feature (default load_fast=true)
+    which prevents multiple plugins from loading correctly.
+    """
+    task_id = os.environ["DET_TASK_ID"]
+    port = os.environ["TENSORBOARD_PORT"]
+
+    tensorboard_args = [
+        "tensorboard",
+        f"--port={port}",
+        f"--path_prefix=/proxy/{task_id}",
+        *add_args,
+    ]
+
+    # Version dependant args
+    version_parts = tb_version.split(".")
+    major = int(version_parts[0])
+    minor = int(version_parts[1])
+
+    if major >= 2:
+        tensorboard_args.append("--bind_all")
+    if major > 2 or major == 2 and minor >= 5:
+        tensorboard_args.append("--load_fast=false")
+
+    tensorboard_args.append(f"--logdir={tfevents_dir}")
```

Comment:
do we lose metadata in tensorboard when we normally would pass muliple logdirs, but now we always only pass one?